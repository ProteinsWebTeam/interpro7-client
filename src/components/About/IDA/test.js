// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import AboutIDA from '.';

const renderer = new ShallowRenderer();

describe('<AboutIDA />', () => {
  test('should render with data', () => {
    renderer.render(<AboutIDA />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
