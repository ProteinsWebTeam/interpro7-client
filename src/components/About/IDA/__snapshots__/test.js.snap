// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutIDA /> should render with data 1`] = `
<React.Fragment>
  <section
    className="banner"
  >
    <div
      className="image-tool-ida"
    />
    <div>
      <h3>
        InterPro Domain Architecture
      </h3>
      The InterPro Domain Architecture (IDA) tool allows you to search the InterPro database with a particular set of domains, and returns all of the domain architectures and associated proteins that match the query.
    </div>
    <Memo(Loadable)
      data={
        {
          "description": "he InterPro Domain Architecture (IDA) tool allows to search the InterPro database with a particular set of domains",
          "name": "About InterPro Domain Architectures",
        }
      }
      processData={[Function]}
    />
  </section>
  <section>
    <div>
      <h3>
        How are Domain architectures derived?
      </h3>
      <p>
        Domain architectures are derived from matches to Pfam Entries. The results will show all proteins matching the criteria selected below. You can select entries which must be included or excluded from your search results. Entries can be selected by either entering a Pfam accession, or an InterPro accession if a Pfam entry is integrated with it.
      </p>
    </div>
    <div>
      <h4>
        How to search for proteins with a specific domain architecture?
      </h4>
      <img
        alt="Add domain"
        className="image-funding"
        src={{}}
      />
      <div
        className="steps-list"
      >
        <ol>
          <li>
            <b>
              Add domain to include
            </b>
             button should be clicked to add a domain to the search.
          </li>
          <li>
            More domains can be added to the search if needed.
          </li>
          <li>
            If the order in which the domains are added is important,
             
            <b>
              Order of domain matters
            </b>
             checkbox should be checked.
            <ol
              style={
                {
                  "listStyleType": "lower-alpha",
                }
              }
            >
              <li>
                When the 
                <i>
                  Order of domains
                </i>
                 option is selected you can drag the domain to the desired position.
              </li>
            </ol>
          </li>
          <li>
            To exclude a particular domain in the search query,
             
            <b>
              Add domain to exclude
            </b>
             button should be clicked to add the domain.
          </li>
          <li>
            If the search results should match the exact domains given, the
             
            <b>
              Exact match
            </b>
             should be checked or the combination of the given domains with any other domain would be shown.
          </li>
          <li>
            At last the proteins with the given domain architecture are shown. By default, Pfam entries are shown. It can be changed to show InterPro entries by toggling the Pfam checkbox to InterPro and vice versa.
          </li>
        </ol>
      </div>
    </div>
  </section>
</React.Fragment>
`;
